### DISCUSSION TITLE
Uploading Multiple Images via selection and/or drag-and-drop

### DISCUSSION THREAD

**Discussion by adfrost81** [netbox-community/netbox#9839] [Ideas]
Uploading images to a page individually via the Attach an Image button is extremely cumbersome. You must do so via a file selection screen and only one at a time. This makes uploading hundreds (or potentially thousands in our case) of images untenable. I'd like to see a couple things implemented to improve this process:

1. Allow selection of multiple files in the file selection dialog.
2. Allow drag-and-drop of single/multiple images from a window or the desktop directly to the Images section on the page.

I love having the ability to attach images to a page, and the display of them is handled well, but it's way too difficult to get them into NetBox in quantity.

**Comment by craigmoscardini** [comment]
This sounds like a great idea. We don't use Netbox to store images for this very reason. Maybe this could be extended further also to allow a text description field against images, or organise into folders/categories.

**Comment by adfrost81** [comment]
Yes, allowing some sort of file heirarchy would be nice too. We're also using the Documents plugin, but it uses the same kind of UI as Images and thus has the same limitations (no drag & drop, no folders, etc). Both features are useful & better than nothing, but far from as useful as they could be.

**Comment by ITCMD** [comment]
+1 on this feature. Uploading 30 site photos one at a time is way too tedius.
